arch/arc/include/swap_macros.h

```diff
@@ -412,12 +412,15 @@
 .macro _store_old_thread_callee_regs
 
 	_save_callee_saved_regs
-#ifdef CONFIG_SMP
```

Comment:
Storing the old thread in the switch handle doesn't sound like belonging to "_store_old_thread_callee_regs". Should it be done elsewhere? Isn't it saved already by '*old_thread =  _current' in z_arch_get_next_switch_handle?
No longer conditional to SMP only?

Reply:
>  Isn't it saved already by '*old_thread = _current' in z_arch_get_next_switch_handle?

No, it isn't. We previously pass old_thread to `z_get_next_switch_handle`  (where it was saved) in `z_arch_get_next_switch_handle`, but now we pass NULL to `z_get_next_switch_handle`.

> No longer conditional to SMP only?
Yes, as we pass NULL to `z_get_next_switch_handle` unconditionally.

> Storing the old thread in the switch handle doesn't sound like belonging to "_store_old_thread_callee_regs"
Indeed the macro name is misleading. But as we don't introduce this behavior here I'll keep this cleanup for interrupt code rework.